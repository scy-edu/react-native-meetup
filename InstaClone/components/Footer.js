'use strict';

import React, {
	Component,
} from 'react';
import {
	View,
	Text,
	StyleSheet,
} from 'react-native';
import PropTypes from 'prop-types';

export const styles = StyleSheet.create({
	default: {
		alignSelf: 'stretch',
		alignItems: 'center',
		justifyContent: 'center',
		borderTopWidth: 0.2,
		borderTopColor: 'grey',
		padding: 10,
	},
	label: {
		fontSize: 20,
		fontFamily: 'Avenir',
	},
});

const Footer = ({
	label,
	type,
}) => (
	<View style={styles[type]}>
		<Text style={styles['label']}>{label}</Text>
	</View>
);

Footer.propTypes = {
	label: PropTypes.string.isRequired,
	type: PropTypes.string.isRequired,
};

Footer.defaultProps ={
	label: 'Built in Los Angeles',
	type: 'default',
};

Footer.displayName = 'Footer';

export default Footer;
