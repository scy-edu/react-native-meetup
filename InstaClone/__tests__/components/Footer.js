'use strict';

import 'react-native';
import React from 'react';
import Footer, {
	styles,
} from '../../components/Footer';

import renderer from 'react-test-renderer';
import {
	shallow,
} from 'enzyme';

const createTestProps = () => ({
	label: 'Built in Los Angeles',
	type: 'default',
});

describe('<Footer />', () => {
	let wrapper;

	describe('rendering', () => {
		// run before each test
		beforeEach(() => {
			wrapper = shallow(<Footer {...createTestProps()} />);
		});

		it('renders correctly', () => {
			const tree = renderer.create(<Footer />).toJSON();
			expect(tree).toMatchSnapshot();
		});

		it('renders Built in Los Angeles in the name', () => {
			expect(wrapper.find('Text').contains('Built in Los Angeles')).toBe(true);
		});

		describe('styling', () => {
			it('should have the title style', () => {
				expect(wrapper.find('Text').prop('style')).toBe(styles['label']);
			});
		});
	});

});
