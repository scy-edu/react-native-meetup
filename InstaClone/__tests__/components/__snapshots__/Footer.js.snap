// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> rendering renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "borderTopColor": "grey",
      "borderTopWidth": 0.2,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "fontFamily": "Avenir",
        "fontSize": 20,
      }
    }
  >
    Built in Los Angeles
  </Text>
</View>
`;
